packages/headless/src/utils/validate-payload.ts

```diff
@@ -41,6 +41,18 @@ export const optionalNonEmptyVersionString = new StringValue({
   regex: /^\d+\.\d+\.\d+$/,
 });
 
+export const optionalTrackingId = new StringValue({
+  required: false,
+  emptyAllowed: false,
+  regex: /^[a-zA-Z0-9_\-.]{1,100}$/,
```

Comment:
Comparing with the [regex in analytics](https://github.com/coveo-platform/analytics_schema/blob/e50dc8f35b59aa681b48ea27ad7e3e409c7b89c3/schemas/common/types.json#L250C26-L251C24) for trackingId makes me think you are missing a `\` in here. That should read:
```suggestion
  regex: /^[a-zA-Z0-9_\-\.]{1,100}$/,
```?